### ISSUE TITLE
Are you considering supporting the java version?

### ISSUE THREAD

**Issue by WangkeqiangS** [binwiederhier/ntfy-android#92]
Our project is old and uses java language, while ntfy is kotlin version, how should we deal with it?

**Comment by deathbarn** [comment]
I'm not sure what you're trying to do, but Java and Kotlin are compatible.

**Comment by WangkeqiangS** [comment]
How can I quickly integrate ntfy functionality into a project?

**Comment by deathbarn** [comment]
I think this will be helpful.
https://docs.ntfy.sh/subscribe/phone/#integrations

**Comment by WangkeqiangS** [comment]
thank you for the response，Our project needs the function of message notification, now we find that ntfy is very suitable for us, and we can also receive messages through the demo you provided, but the problem now is that I don't know how to integrate the functions of ntfy into my own project, may I ask whether this is possible?

**Comment by deathbarn** [comment]
I think you'll need to refer to the ntfy source code and API documentation to implement topic subscriptions and notification handling.
https://docs.ntfy.sh/subscribe/api/

**Comment by WangkeqiangS** [comment]
Can I understand that by copying ntfy source code into our project, we can use ntfy functionality?

**Comment by deathbarn** [comment]
That’s correct. Upon further search, I found a library that allows using ntfy with Java. I think this could be useful.
https://github.com/MaheshBabu11/ntfy-java/

**Comment by WangkeqiangS** [comment]
Thanks, I am an Android project, this library is server-side java, may not be suitable for me. I'm a little confused, nfty's Android source code is a bit excessive, how do I copy the source code into my own project?

**Comment by deathbarn** [comment]
Sorry, I just realized it’s a server-side library. If there’s no client-side library available, the only option I can think of is to copy and implement it manually.

**Comment by WangkeqiangS** [comment]
thank you very much，But the problem is that ntfy has a lot of source code, and there are few comments in the demo, I want to ask where to copy the source code?

**Comment by deathbarn** [comment]
I'm not very familiar with ntfy, so I might be off the mark.
```
app/src/main/java/io/heckel/ntfy/service/*
app/src/main/java/io/heckel/ntfy/msg/ApiService.kt
```

**Comment by WangkeqiangS** [comment]
Do you mean to copy from the code under the service package first?

**Comment by p1gp1g** [comment]
If you want to add push notifications to an application for a system without Google Services, you may want to follow https://unifiedpush.org/ (Doc for the next release of the library, soon published: https://unifiedpush.org/@dev/kdoc/)
